E2EReporter.ts

```diff
@@ -0,0 +1,207 @@
+  suites: CustomSuiteResult[];
+  totalDuration: number;
+  totalTests: number;
+  testsPassed: number;
+  testsFailed: number;
+  testsSkipped: number;
+  testsTimedout: number;
+  testsFlaky: number;
+  averageTestDuration: number;
+  endTime: Date;
+  status: string;
+  environment: string;
+  failures: CustomTestResult[];
+  skipped: CustomTestResult[];
+  flakes: CustomTestResult[];
+}
+
+class E2EReporter implements Reporter {
+  private suites: CustomSuiteResult[] = [];
+  private startTime: number | undefined;
+  private endTime: number | undefined;
+  private tempData = {
+    totalTests: 0,
+    totalCompleted: 0,
+    testsPassed: 0,
+    testsFailed: 0,
+    testsSkipped: 0,
+    testsTimedout: 0,
+    testsFlaky: 0,
+    duration: 0,
+    numberOfWorkers: 0,
+    failures: [] as CustomTestResult[],
+    skipped: [] as CustomTestResult[],
+    flakes: [] as CustomTestResult[],
+  };
+
+  onBegin() {
+    this.startTime = Date.now();
+  }
+
+  onTestBegin(test: TestCase) {
+    const suiteId = test.parent.title.split(" ").join(" ");
+    const suiteResult = this.suites.find((s) => s.suiteId === suiteId);
+
+    const data: CustomSuiteResult = {
+      title: test.parent.title,
+      fileName: test.parent.parent!.title,
+      duration: 0,
+      tests: [],
+      suiteId: suiteId,
+    };
+
+    if (!suiteResult) {
+      this.suites.push(data);
+    }
+  }
+
+  onTestEnd(test: TestCase, result: TestResult) {
+    // find the suite this test belongs to
+    const suiteId = test.parent.title.split(" ").join(" ");
+    const suiteResult = this.suites.find((s) => s.suiteId === suiteId);
+
+    // @ts-ignore - using _projectId as projectId is not in the testCase types
+    const browserName = test._projectId as string;
+
+    // build test data
+    const data: CustomTestResult = {
+      title: test.title,
+      status: result.status,
+      duration: result.duration,
+      retries: result.retry,
+      errors: result.errors,
+      browser: browserName,
+    };
+
+    // add this to test to its suite
+    suiteResult?.tests.push(data);
+
+    // build test stats
+    switch (result.status) {
+      case "passed":
+        this.tempData.testsPassed++;
+        break;
+
+      case "failed":
+        this.tempData.testsFailed++;
+        this.tempData.failures.push(data);
+        break;
+
+      case "timedOut":
+        this.tempData.testsTimedout++;
+        this.tempData.failures.push(data);
+        break;
+
+      case "interrupted":
+        this.tempData.testsTimedout++;
+        this.tempData.failures.push(data);
+        break;
+
+      case "skipped":
+        this.tempData.testsSkipped++;
+        this.tempData.skipped.push(data);
+        break;
+
+      default:
+        this.tempData.testsFlaky++;
+        this.tempData.flakes.push(data);
+        break;
+    }
+
+    // increase test count
+    this.tempData.totalTests++;
+
+    // add duration
+    this.tempData.duration = this.tempData.duration + result.duration;
```

Comment:
Not sure if this is any better:
```suggestion
    this.tempData.duration += result.duration;
```